cordova-lib/src/configparser/ConfigParser.js

```diff
@@ -279,113 +269,108 @@ ConfigParser.prototype = {
 
         return scriptElements.filter(filterScriptByHookType);
     },
-
+   /**
+    * Returns a list of plugin (IDs)
+    * @return {string[]} Array of plugin IDs
+    */
+    getPluginIdList: function () {
+        var plugins = this.doc.findall('plugin');
+        var result = plugins.map(function(plugin){
+            return plugin.attrib.name;
+        });
+        var features = this.doc.findall('feature');
+        features.forEach(function(element ){
+            var idTag = element.find('./param[@name="id"]');
+            if(idTag){
+                result.push(idTag.attrib.value);
+            }
+        });
+        return result;
+    },
     /**
-     * Returns a list of features (IDs)
-     * @return {string[]} Array of feature IDs
+     * Adds a plugin element. Does not check for duplicates.
+     * @name addPlugin
+     * @function
+     * @param {object} attributes name, version and src are supported
+     * @param {Array} variables name, value
      */
-    getFeatureIdList: function () {
-        var features = this.doc.findall('feature'),
-            feature, idTag, id,
-            result = [];
-
-        // Check for valid features that have IDs set
-        for (var i = 0, l = features.length; i < l; ++i) {
-            feature = features[i];
-            idTag = feature.find('./param[@name="id"]');
-            if (null === idTag) {
-                // Invalid feature
-                continue;
-            }
-            id = idTag.attrib.value;
-            if (!!id) {
-                // Has id and id is non-empty
-                result.push(id);
-            }
+    addPlugin: function(attributes, variables){
+        if ( !attributes && !attributes.name ) return;
```

Comment:
Why silently fail here ? shouldn't we throw instead ? the client of this module might think that everything is fine, when in fact, the changes he planned haven't gone through.